engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/parser/BpmnParse.java

```diff
@@ -3566,7 +3566,7 @@ protected void parseIntermediateTimerEventDefinition(Element timerEventDefinitio
 
     Element timeCycleElement = timerEventDefinition.element("timeCycle");
     if (timeCycleElement != null) {
-      addTimeCycleWarning(timeCycleElement, "intermediate catch", timerActivity.getId());
+      LOG.intermediateCatchTimerEventWithTimeCycleNotRecommended(timerActivity.getId());
```

Comment:
It will be helpful to add the `processDefinitionKey` to the message to be easier for the user to distinguish to which process the message is meant for. Then the message can look like `definitionKey: {}; It is not recommended to use a intermediate catch timer event with a time cycle, element with id '{}'.`
```suggestion
      ProcessDefinition processDefinition = (ProcessDefinition) timerActivity.getProcessDefinition();
      LOG.intermediateCatchTimerEventWithTimeCycleNotRecommended(processDefinition.getKey(), timerActivity.getId());
```